test/tests/functional/pbs_test_tpp.py

```diff
@@ -386,30 +431,412 @@ def test_multiple_comm_with_mom(self):
+        leaf_val = self.hostC + "," + self.hostA + "," + self.hostE
+        b = {'PBS_LEAF_ROUTERS': leaf_val}
+        self.set_pbs_conf(host_name=self.hostB, conf_param=b)
+        leaf_val = self.hostE + "," + self.hostC + "," + self.hostA
+        b = {'PBS_LEAF_ROUTERS': leaf_val}
+        self.set_pbs_conf(host_name=self.hostD, conf_param=b)
+        set_attr = {ATTR_l + '.select': '3:ncpus=1',
+                    ATTR_l + '.place': 'scatter', ATTR_k: 'oe'}
+        resv_set_attr = {ATTR_l + '.select': '3:ncpus=1',
+                         ATTR_l + '.place': 'scatter',
+                         'reserve_start': int(time.time()) + 30,
+                         'reserve_end': int(time.time()) + 120}
+        self.common_steps(set_attr=set_attr, resv_set_attr=resv_set_attr,
+                          job=True, interactive=True, resv=True,
+                          resv_job=True)
+
+    @requirements(num_moms=2, num_comms=2)
+    def test_COMM_ROUTERS_with_nondefault_pbs_conf(self):
+        """
+        This test verifies communication between server-mom and
+        between mom when multiple pbs_comm are present in cluster
+        when PBS_COMM_ROUTERS has ipaddress, port number and hostname
+        as values and pbs.conf is in non default location
+        Configuration:
+        Node 1 : Server, Sched, Mom, Comm
+        Node 2 : Mom
+        Node 3 : Comm
+        """
+        self.common_setup()
+        hostA_ip = socket.gethostbyname(self.hostA)
+        comm_val = self.hostA + ":17001"
+        a = {'PBS_COMM_ROUTERS': self.hostA}
+        self.set_pbs_conf(host_name=self.hostC, conf_param=a)
+        comm_val = hostA_ip + ":17001" + "," + self.hostC
+        a = {'PBS_COMM_ROUTERS': comm_val}
+        self.set_pbs_conf(host_name=self.hostE, conf_param=a)
+        leaf_val = self.hostA + "," + self.hostC + "," + self.hostE
+        b = {'PBS_LEAF_ROUTERS': leaf_val}
+        self.set_pbs_conf(host_name=self.hostA, conf_param=b)
+        leaf_val = self.hostC + "," + self.hostA + "," + self.hostE
+        b = {'PBS_LEAF_ROUTERS': leaf_val}
+        self.set_pbs_conf(host_name=self.hostB, conf_param=b)
+        leaf_val = self.hostE + "," + self.hostC + "," + self.hostA
+        b = {'PBS_LEAF_ROUTERS': leaf_val}
+        self.set_pbs_conf(host_name=self.hostD, conf_param=b)
+        self.copy_pbs_conf_to_non_default_path()
+        set_attr = {ATTR_l + '.select': '3:ncpus=1',
+                    ATTR_l + '.place': 'scatter', ATTR_k: 'oe'}
+        resv_set_attr = {ATTR_l + '.select': '3:ncpus=1',
+                         ATTR_l + '.place': 'scatter',
+                         'reserve_start': int(time.time()) + 30,
+                         'reserve_end': int(time.time()) + 120}
+        self.common_steps(set_attr=set_attr, resv_set_attr=resv_set_attr,
+                          job=True, interactive=True, resv=True,
+                          resv_job=True)
+
     def tearDown(self):
         TestFunctional.tearDown(self)
```

Comment:
TestFunctional.tearDown() call should be at the last line of custom tearDown() method.